.github/workflows/ci.yaml

```diff
@@ -57,16 +59,48 @@ jobs:
         run: |
           composer update --no-interaction --prefer-dist --no-progress ${{ matrix.dependencies }}
 
-      - name: Run tests using Makefile
-        if: runner.os != 'Windows'
+      - name: Run tests
+        shell: bash
         run: |
           make test-unit
 
-      - name: Run tests without Makefile
+      - name: Run integration tests
+        shell: bash
+        run: |
+          make test-unit PHPUNIT_GROUP=integration
+
+      - name: Cache E2E tests dependencies
         if: runner.os == 'Windows'
+        uses: actions/cache@v2
+        with:
+          path: tests/e2e/*/vendor
+          key: e2e-vendor-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('tests/e2e/*/composer.json') }}-${{ matrix.dependencies }}
+          restore-keys: |
+            e2e-vendor-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('tests/e2e/*/composer.json') }}-
+            e2e-vendor-${{ runner.os }}-${{ matrix.php-version }}-
+            e2e-vendor-${{ runner.os }}-
+            composer-
```

Comment:
Installing anything with Composer is insanely slow in Windows, just compare [these 4 seconds on Linux](https://github.com/infection/infection/pull/1365/checks?check_run_id=1296609944#step:6:1) with [these 34 seconds on Windows](https://github.com/infection/infection/pull/1365/checks?check_run_id=1296609961#step:6:1). The solution is to cache the dependencies, but not only archives, the whole `vendor` folders. Note that we don't cache `composer.lock` here: it takes far less time to verify that all is in order.

This might be a mistake, but let's see.